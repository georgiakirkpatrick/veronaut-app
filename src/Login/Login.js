import React, { useState } from 'react'
import { Link, useHistory } from 'react-router-dom'
import AuthApiService from '../services/auth-api-service'
import FormTitle from '../FormTitle/FormTitle'
import FormButton from '../FormButton/FormButton'
import FormTextInput from '../FormTextInput/FormTextInput'
import TokenService from '../services/token-service'
import './Login.css'

const Login = props => {
    const {
        routeProps
    } = props

    const [error, setError] = useState(null)
    const [loginEmail, setLoginEmail] = useState('')
    const [loginPassword, setLoginPassword] = useState('')

    const history = useHistory()

    const advancePage = () => {
        console.log('routeProps', routeProps)

        const referrer = routeProps.location.state
            ? routeProps.location.state.referrer
            : '/'

        history.push(referrer)
    }

    // const handleSubmitJwtAuth = event => {
    //     event.presentDefault()

    //     AuthApiService.postLogin({
    //         email: loginEmail,
    //         password: loginPassword
    //     })
    //     .then(response => {
    //         setLoginEmail('')
    //         setLoginPassword('')
    //         TokenService.saveAuthToken(response.authToken)
    //     })
    //     .catch(response => {
    //         setError(response.error)
    //     })
    // }

    // const saveAuthToken = (email, password) => {
    //     const authToken = TokenService.makeBasicAuthToken(email, password)
        
    //     TokenService.saveAuthToken(authToken)
    // }

    const handleLoginSubmit = event => {
        if (loginEmail === '' && loginPassword === '') {
            alert('Please enter your email and password')
        } else if (loginEmail === '' && loginPassword !== '') {
            alert('Please enter your email')
        } else if (loginEmail !== '' && loginPassword === '') {
            alert('Please enter your password')
        } else {
            event.preventDefault()

            AuthApiService.postLogin({
                email: loginEmail,
                password: loginPassword
            })
            .then(response => {
                console.log('response', response)
                setLoginEmail('')
                setLoginPassword('')
                TokenService.saveAuthToken(response.authToken)
            })
            .catch(response => {
                setError(response.error)
                console.log('error', error)
            })

            advancePage()
        }
    }

    return (
        <form className='Login'>
            <header>
                <FormTitle titleText='Log in' />
            </header>
            
            <fieldset className='Login__fieldset'>
                <FormTextInput
                    id='login-email'
                    prompt='Email'
                    name='loginEmail'
                    currentValue={loginEmail}
                    handleChange={event => {setLoginEmail(event.target.value)}}
                />

                <FormTextInput
                    id='login-password'
                    prompt='Password'
                    name='loginPassword'
                    currentValue={loginPassword}
                    handleChange={event => {setLoginPassword(event.target.value)}}
                />

                <Link to='/forgot-password'>Forgot password?</Link>
            </fieldset>

            <FormButton 
                buttonText='SIGN IN'
                handleClick={event => handleLoginSubmit(event)}
            />

            <Link to={`/create-account`} className='main-link'>
                Create an account
            </Link>
        </form>
    )
}

export default Login