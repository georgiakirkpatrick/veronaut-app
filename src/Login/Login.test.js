import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import Login from './Login'
import { handleLoginSubmit } from './Login'

describe ('Login', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Router><Login /></Router>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Router><Login /></Router>)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })

    // it('checks that all required fields are supplied', () => {
    //     const loginEmail = 'test@email.com'
    //     const loginPassword = 'testpassword'

    //     handleLoginSubmit()
    // })
})