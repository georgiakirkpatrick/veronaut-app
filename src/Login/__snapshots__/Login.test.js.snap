// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders the UI as expected 1`] = `
<form
  className="Login"
>
  <header>
    <h2
      className="FormTitle"
    >
      Log in
    </h2>
  </header>
  <fieldset
    className="Login__fieldset"
  >
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="login-email"
      >
        Email
      </label>
      <input
        id="login-email"
        name="loginEmail"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="login-password"
      >
        Password
      </label>
      <input
        id="login-password"
        name="loginPassword"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <a
      href="/forgot-password"
      onClick={[Function]}
    >
      Forgot password?
    </a>
  </fieldset>
  <button
    className="FormButton"
    onClick={[Function]}
    type="button"
  >
    SIGN IN
  </button>
  <a
    className="main-link"
    href="/create-account"
    onClick={[Function]}
  >
    Create an account
  </a>
</form>
`;
